runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -1059,12 +1114,15 @@ TR_J9InlinerPolicy::genCodeForUnsafeGetPut(TR::Node* unsafeAddress,
       debugTrace(tracer(), "\t In genCodeForUnsafeGetPut, Block %d created for isClass Test\n", isClassBlock->getNumber());
       }
 
-   cfg->removeEdge(beforeCallBlock, indirectAccessBlock);
+   if (indirectAccessBlock)
+      cfg->removeEdge(beforeCallBlock, indirectAccessBlock);
+
+   if (arrayCheckNeeded && arrayBlockNeeded)
+      cfg->removeEdge(nullComparisonBlock, arrayDirectAccessBlock);
 
-   //Generating treetop and block for array check
    if (needNullCheck)
       {
-      TR::TreeTop *treeBeforeCmp = firstComparisonTree->getPrevTreeTop();
+      TR::TreeTop *treeBeforeCmp = firstComparisonTree ? firstComparisonTree->getPrevTreeTop() : beforeCallBlock->getExit()->getPrevTreeTop();
```

Comment:
In case (4) the block won't be split and `firstComparisonTree` will be a null pointer.  Will this end up placing the `NULLCHK` after the point where the call originally appeared?

Reply:
In case 4, the `NULLCHK` ends up being placed like so: 
```
n1n       BBStart <block_2>                                                                   [    0x7b404f004660] bci=[-1,0,33] rc=0 vc=14 vn=- li=- udi=- nc=0
n42n      NULLCHK on n3n [#32]                                                                [    0x7b404f005330] bci=[-1,5,33] rc=0 vc=0 vn=- li=- udi=- nc=1
n41n        PassThrough                                                                       [    0x7b404f0052e0] bci=[-1,5,33] rc=1 vc=0 vn=- li=- udi=- nc=1
n3n           aload  TestUnsafe.unsafe Lsun/misc/Unsafe;[#405  Static] [flags 0x307 0x0 ]     [    0x7b404f004700] bci=[-1,0,33] rc=1 vc=14 vn=- li=- udi=- nc=0
n44n      astore  <temp slot 4>[#418  Auto] [flags 0x7 0x0 ]                                  [    0x7b404f0053d0] bci=[-1,3,33] rc=0 vc=0 vn=- li=- udi=- nc=1
n4n         aload  <parm 0 [I>[#403  Parm] [flags 0x40000107 0x0 ]                            [    0x7b404f004750] bci=[-1,3,33] rc=1 vc=14 vn=- li=- udi=- nc=0
n46n      lstore  <temp slot 5>[#419  Auto] [flags 0x4 0x0 ]                                  [    0x7b404f005470] bci=[-1,4,33] rc=0 vc=0 vn=- li=- udi=- nc=1
n5n         lload  <parm 1 J>[#404  Parm] [flags 0x40000104 0x0 ]                             [    0x7b404f0047a0] bci=[-1,4,33] rc=1 vc=14 vn=- li=- udi=- nc=0
n72n      BBEnd </block_2> =====                                                              [    0x7b404f0a0640] bci=[-1,5,33] rc=0 vc=0 vn=- li=- udi=- nc=0
```
I believe that's the expected behaviour here...but of course, if I'm mistaken then please let me know and I'll fix it.

Reply:
Oh, I see.  I missed the fact that the block actually is being split in case (4) here:  https://github.com/eclipse-openj9/openj9/blob/b42906eb7859c07ba5782a02e8dd56dbec96aaad/runtime/compiler/optimizer/InlinerTempForJ9.cpp#L946-L953